File: udemy/ds-and-a/DSAndA.Test/LinkedLists/Assignment1/AppendLastNodesToBeginning.Test.cs
using DSAndA.LinkedLists;
using Xunit;

namespace DSAndA.Test.LinkedLists.Assignment1
{
    public class AppendLastNodesToBeginningTestFixture
    {
        [Fact]
        public void AppendLastNodesToBeginningTest1()
        {
            LinkedList<int> linkedList = new LinkedList<int>();
            linkedList.InsertAtHead(3);
            linkedList.InsertAtHead(2);
            linkedList.InsertAtHead(1);

            linkedList.AppendLastNodesToBeginning(n: 1);

            Assert.Equal("3,1,2,", linkedList.ToString());
        }

        [Fact]
        public void AppendLastNodesToBeginningTest2()
        {
            LinkedList<int> linkedList = new LinkedList<int>();
            linkedList.InsertAtHead(4);
            linkedList.InsertAtHead(3);
            linkedList.InsertAtHead(2);
            linkedList.InsertAtHead(1);

            linkedList.AppendLastNodesToBeginning(n: 2);

            Assert.Equal("3,4,1,2,", linkedList.ToString());
        }
    }
}

File: udemy/ds-and-a/javascript-solutions/src/tests/LinkedLists/Assignment1/AppendLastNodesToBeginning.test.js
const LinkedList = require("../../../LinkedLists/LinkedList");

test('Append last N nodes to beginning', () => {
    let linkedList = new LinkedList();
    linkedList.insertAtHead(4);
    linkedList.insertAtHead(3);
    linkedList.insertAtHead(2);
    linkedList.insertAtHead(1);
    linkedList.insertAtHead(0);

    expect(linkedList.toString()).toBe('0, 1, 2, 3, 4');

    const n = 2;
    linkedList.appendLastNodesToBeginning(2);

    expect(linkedList.toString()).toBe('3, 4, 0, 1, 2');
});